tiledb/sm/filesystem/posix.h

```diff
@@ -177,7 +175,7 @@ class Posix {
    * @param new_path The new path.
    * @return Status
    */
-  Status move_path(const std::string& old_path, const std::string& new_path);
+  Status move_file(const URI& old_path, const URI& new_path) const override;
```

Comment:
We should have documentation for all of these two-URI functions that they must exist on the same filesystem. The only reason this isn't a preexisting problem is that we're passing URI in now and not raw paths.
It would probably be prudent to validate the arguments this way, but that would be out of scope here.